### ISSUE TITLE
Error when trying to predict sparse lil matrix format with my auto sklearn model (trained with scipy sparse csr matrix)

### ISSUE THREAD

**Issue by krzischp** [automl/auto-sklearn#1034]
## Describe the bug ##
With SKlearn models, there is no problem:
``` python
X_train = scipy.sparse.csr_matrix(X_train.values)
X_test = scipy.sparse.csr_matrix(X_test.values)

svm = sklearn.svm.SVC(kernel='rbf', probability=True)
svm.fit(X_train, y_train)

## Conv to lil_matrix
X_train, X_test, y_train, y_test = train_test_split(
X, y, test_size=0.20, random_state=42, stratify=y)

X_test = scipy.sparse.lil_matrix(X_test.values)
# prediction
svm.predict_proba(X_test)
```
gives the expected result with no error.

But if I try with my Auto Sklearn trained model:

``` python
## model trained with scipy.sparse.csr_matrix dataset
pickle_in = open("automlAutoSklearn.pickle","rb")
autosk = pickle.load(pickle_in)
pickle_in.close()

X_test = scipy.sparse.lil_matrix(X_test.values)
autosk.predict_proba(X_test)
``` 
Then I get the following log:

``` 
ValueError                                Traceback (most recent call last)
<ipython-input-7-ce0445ddb3f8> in <module>
      3 
      4 X_test = scipy.sparse.lil_matrix(X_test.values)
----> 5 autosk.predict_proba(X_test)
<...>
ValueError: Auto-sklearn previously received features of type <class 'scipy.sparse.csr.csr_matrix'> yet the current features have type <class 'scipy.sparse.lil.lil_matrix'>. Changing the dtype of inputs to an estimator is not supported.
``` 

Because of this issue, auto sklearn is not working with the library shap, when the auto-sklearn models have been trained 
with scipy.sparse.csr_matrix.
https://github.com/slundberg/shap

Indeed, the shap explainer uses both formats (scipy.sparse.csr_matrix and scipy.sparse.lil.lil_matrix) when executing its algorithm that calculates the shap values:
 
this code
``` python
X_train = scipy.sparse.csr_matrix(X_train.values)
X_test = scipy.sparse.csr_matrix(X_test.values)
# use Kernel SHAP to explain test set predictions
explainer = shap.KernelExplainer(autosk.predict_proba, X_train, link="logit")
shap_values = explainer.shap_values(X_test, nsamples=100)
``` 
Gives the same **ValueError** issue.

## Expected behavior ##
I would expect that it gives no error like when predicted with the Scikit-Learn model.

## Actual behavior, stacktrace or logfile ##
``` ValueError: Auto-sklearn previously received features of type <class 'scipy.sparse.csr.csr_matrix'> yet the current features have type <class 'scipy.sparse.lil.lil_matrix'>. Changing the dtype of inputs to an estimator is not supported.```

## Environment and installation: ##

* Linux
* conda environment
* Python version: 3.6
* Auto-sklearn version: 0.11.1

**Comment by mfeurer** [comment]
Thanks a lot for reporting this. I can confirm that this is a bug. We'll try to put out a fix as soon as possible, but I'm afraid the earliest will be mid January.

Could you simply convert the sparse matrix to circumvent the issue in the meantime?

**Comment by mfeurer** [comment]
Sorry, I missed closing this. It should be fixed as of version 0.12.2 from the very beginning of this year. Please reopen if that's not the case.
